src/components/lifecycle/lifecycle.ts

```diff
@@ -0,0 +1,110 @@
+import { LitElement, html } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import Styles from './lifecycle.scss';
+import '../button/button';
+import { BUTTON_SIZES } from '../button/defs';
+
+@customElement('kd-lifecycle')
+export class LitLifecycle extends LitElement {
+  @property({ type: Number }) counter = 0;
+
+  /** Specifies the size of the button. */
+  @property({ type: String })
+  size: BUTTON_SIZES = BUTTON_SIZES.SMALL;
+
+  @property({ type: String })
+  primaryBtnText = '';
+
+  @property({ type: String })
+  secondaryBtnText = '';
+
+  static override styles = Styles;
+  constructor() {
+    super();
+    console.log('Constructor: Component is being created');
+    this.counter = 0; // Initial property value
```

Comment:
This is redundant, as you've already initialized the `counter` value as a `@property` above.